docs/docs/aleph_zero/sdk.md

```diff
@@ -0,0 +1,386 @@
+
+### Structure
+
+Invariant SDK consists of 3 separate contracts: our DEX contract (Invariant), token contract (PSP22), contract that can be used to wrap native currency (Wrapped AZERO), various values and helper functions. You can deploy your own or use an existing contract.
+
+### Txs and Queries
+
+After you load or deploy a contract you can start interacting with it. In order to make a contract call you have to use methods of class that represents specific contract. First parameter will always be an account you want to use and rest of parameters will be an entrypoint parameters.
+
+### Event listeners
+
+Invariant class has additional methods that allows you to attach event listeners so you can run your own code based on contract activity.
+
+### Values and Helper functions
+
+SDK also contains all essential values and helper functions that might be needed in your app like maximum tick, maximum price, calculating price impact and many others that will simplify your calculations.
+
+## Files
+
+```
+📦sdk
+ ┣ 📂contracts
+ ┃ ┣ 📂invariant
+ ┃ ┣ 📂PSP22
+ ┃ ┗ 📂wrapped-azero
+ ┣ 📂math
+ ┣ 📂src
+ ┗ 📂tests
+```
+
+### Contracts
+
+Contracts folder contains deploy-ready contracts and metadata that is used to interact with them.
+
+### Math
+
+Math contains Rust structs and data from main Invariant contract exported using WASM.
+
+### Source
+
+Everything combined in easy-to-use interface.
+
+### Tests
+
+Tests folder with tests to make sure everything works right.
+
+## Usage
+
+### Starting off
+
+```typescript
+const main = async () => {
+  const api = await initPolkadotApi(Network.Main) // initialize api, use enum to specify the network
+
+  // initialize account, you can use your own wallet by pasting mnemonic phase
+  const keyring = new Keyring({ type: 'sr25519' })
+  const account = await keyring.addFromUri('//Alice')
+}
+
+main()
+```
+
+### Initialize DEX and tokens
+
+:::note Why "n" is at the end of every number
+Notice how we use "n" at the end of every number. "n" indicates that specified value is a BigInt, number with higher precision. Read more about BigInt here: [BigInt MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).
+:::
+
+:::info Creating types
+You can create custom decimals with `toDecimal` syntax. First argument is an integer and second argument is a number of zeros. For example `toFee(3n, 2n)` will produce decimal equal to `0.03`. Read more about types here: [Types](types.md).
+:::
+
+:::warning Token sorting
+Tokens are sorted alphabetically when pool key is created, so make sure that you swap tokens in correct direction. Read more about pool keys here: [PoolKey](storage#poolkey).
+:::
+
+```typescript
+const main = async () => {
+  // --snip--
+
+  // load invariant contract
+  const invariant = await Invariant.load(api, Network.Main, INVARIANT_ADDRESS)
+
+  // load token contracts
+  const token0 = await PSP22.load(api, Network.Main, TOKEN0_ADDRESS) // token we want to give
```

Comment:
Require update. interface of psp22 was changed. Now we use single instance of psp22 to interact with tokens